src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1705,6 +1709,29 @@ internal static unsafe bool TryParseDouble(ReadOnlySpan<char> value, NumberStyle
                 {
                     result = double.NaN;
                 }
+                else if (valueTrim.StartsWith(info.PositiveSign, StringComparison.OrdinalIgnoreCase))
+                {
+                    valueTrim = valueTrim.Slice(info.PositiveSign.Length);
+
+                    if (valueTrim.EqualsOrdinalIgnoreCase(info.PositiveInfinitySymbol))
+                    {
+                        result = double.PositiveInfinity;
+                    }
+                    else if (valueTrim.EqualsOrdinalIgnoreCase(info.NaNSymbol))
+                    {
+                        result = double.NaN;
+                    }
+                    else
+                    {
+                        result = 0;
+                        return false;
+                    }
+                }
```

Comment:
I see you called out in your PR description:
"We did not support +Infinity, +NaN, or -NaN in either the UTF8 or UTF16 parsing logic"
but it's not clear to me why that's necessary to support.  What's driving this change / why are these additional cases needed?

Reply:
These are cases that the IEEE 754 specification defines as required for correct handling of floating-point operations. This is part of the push to ensure all of our operations are fully IEEE compliant (as to all required/non-optional parts).
* This included fixing up the parser/formatter to return correct results and to handle the specific edge cases

Reply:
To clarify, we've had numerous bugs logged for parsing/formatting issues with the floating-point types, so this is part of the effort to address that.

https://github.com/dotnet/coreclr/issues/19802 tracks all the individual points that the IEEE spec requires for conversion to/from a string, and the following are some of the bugs we are/were tracking (with more that were closed in 2.1, but that I didn't lookup):
* #16783
* #3313
* #13106
* #1316
* #13615
* #17467